image_creator/os_type/linux.py

```diff
@@ -538,6 +598,34 @@ def _do_collect_metadata(self):
         else:
             self.out.warn("OpenSSH Daemon is not configured to run on boot")
 
+        if self.is_enabled('cloud-init'):
+            self.cloud_init = True
+        else:
+            # Many OSs use a systemd generator for cloud-init:
+            #
+            # When booting under systemd, a generator will run that determines
+            # if cloud-init.target should be included in the boot goals.
+
+            generator_found = False
+            for i in ("/run", "/etc", "/usr/local/lib", "/usr/lib"):
+                if self.image.g.is_file("%s/systemd/system-generators/"
+                                        "cloud-init-generator" % i):
+                    generator_found = True
+                    break
+            # By default, this generator will enable cloud-init. It will not
+            # enable cloud-init if either:
+            #
+            #   * A file exists: /etc/cloud/cloud-init.disabled
+            #   * The kernel command line as found in /proc/cmdline contains
+            #     cloud-init=disabled. When running in a container, the kernel
+            #     command line is not honored, but cloud-init will read an
+            #     environment variable named KERNEL_CMDLINE in its place.
```

Comment:
Those are directly taken from the cloud-init documentation: http://cloudinit.readthedocs.io/en/latest/topics/boot.html?highlight=systemd#generator

I won't rephrase it but I'll add a link